lib/wanda/executions/server.ex

```diff
@@ -34,6 +34,14 @@ defmodule Wanda.Executions.Server do
       |> Target.get_checks_from_targets()
       |> Catalog.get_checks()
 
+    checks_ids = Enum.map(checks, & &1.id)
+
+    # Remove non-existent checks from targets
+    targets =
+      Enum.map(targets, fn %{checks: target_checks} = target ->
+        %Target{target | checks: intersection(target_checks, checks_ids)}
```

Comment:
It looks a bit strange because the linter removes the brackets because of the right-associativity. Maybe it's clearer as
```
A -- (A -- B)
```

So you are first getting the difference between the two sets/lists, and then removing the difference from one of the sets to get the intersection of the two.

Reply:
@jamie-suse in my opinion the idiomatic way to do this is to use `Enum.filter`, even if using `--` looks more elegant it looks a little bit more opaque and hard to read.
Also, it's a bit weird that this module contains a function that does generic list operations, while the `Enum` standard library module already provides a way to do this.

I would do what @nelsonkopliku  suggested, or if you prefer:
 ```elixir
Enum.filter(target_checks, fn check -> check in check_ids end)
```

Reply:
That said, @jamie-suse function is faster but I just find it hard to read, so I would vouch for clarity in this case as we have very small lists.